include/eeprom/core/hardware_iface.hpp

```diff
@@ -3,12 +3,28 @@
 namespace eeprom {
 namespace hardware_iface {
 
+enum EEPromAddressType {
+    EEPROM_ADDR_8_BIT = sizeof(uint8_t),
+    EEPROM_ADDR_16_BIT = sizeof(uint16_t)
+};
+
+enum EEpromMemorySize { MICROCHIP_256Byte = 256, ST_16KByte = 16384 };
 /**
  * Interface to eeprom. Must be implemented in FW and Simulation
+ *
+ * Consecutive write requests will want to have write protection disabled. The
+ * enable should happen when all the write transactions have completed.
+ *
  */
 class EEPromHardwareIface {
   public:
     EEPromHardwareIface() = default;
+    EEPromHardwareIface(const size_t addr_bytes)
+        : eeprom_addr_bytes(addr_bytes) {
```

Comment:
i'd love to see an init-list that explicitly specifies all the values based on the argument, ideally by using a `switch` statement so the compiler yells at us if we add enum elements and don't handle it